android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/PackageActivity.java

```diff
@@ -238,6 +240,28 @@ private void showErrorToast(Context context, String message) {
     cleanup();
   }
 
+  private boolean loadWelcomePage(List<Map<String, String>> theInstalledKeyboards)
+  {
+    boolean _found=false;
+     for(Map<String,String> _keyboard:theInstalledKeyboards) {
+       String _customlink = _keyboard.get(KMManager.KMKey_CustomHelpLink);
+       if (_customlink != null) {
+         webView.loadUrl("file:///" + _customlink);
+         _found=true;
+         break;
+       }
+     }
+     if(!_found)
+       return false;
+
+      final Button installButton = (Button) findViewById(R.id.installButton);
+      final Button cancelButton = (Button) findViewById(R.id.cancelButton);
+      installButton.setVisibility(View.GONE);
+      cancelButton.setText("OK");
```

Comment:
ah, originally the webView had two buttons for "install" and "cancel".
But for displaying welcome.htm we'll only need a single "OK" button.

Can you find a way to update the layout for this 
`/android/KMAPro/kMAPro/src/main/res/layout/activity_package_installer.xml`

so we get the 2 buttons on readme.htm and 1 button for welcome.htm?

Maybe define 2 linearlayouts and change the visibility depending on which .htm file is displayed.

Reply:
I already hide the install button for the welcome page.

Reply:
> cancelButton.setText("OK");

right, but we probably don't want to maintain a cancel button that has a text of "OK". It leads to questions what happens when you click "OK"?

I'm suggesting in the layout to have 1 linearlayout that has the original 2 buttons. followed by another linearlayout that has a single button ("OK"). Then packageActivity would toggle the visibility of which linear layout gets displayed.

----

Also, you'll want to use the string `label_ok` defined in 
`android\KMEA\app\src\main\res\values\strings.xml`.

e.g. `getString(R.string.label_ok)`

This way (someday) we can localize the app.